pages/instructorQuestions/instructorQuestions.ejs

```diff
@@ -1,136 +1,228 @@
+                <button name="__action" value="add_question" class="btn btn-sm btn-light"><i class="fa fa-plus" aria-hidden="true"></i>
+                  <span>Add question</span></button>
+              </form>
+              <% } %>
             </div>
-            <script>
-                $(function() {
-                    $("#questionsTable").tablesorter({
-                        theme: "bootstrap",
-                        widthFixed: true,
-                        headerTemplate: '{content} {icon}',
-                        widgets: ["uitheme", "filter", "zebra"],
-                        widgetOptions: {
-                            zebra: ["even", "odd"],
-                            filter_reset: "#resetFilter",
-                            filter_cssFilter: "form-control",
-                            filter_functions: {
-                                3: {
-                                    <% all_tags.forEach(function(tag) { %>
-                                    "<%= escapeRegExp(tag.name) %>": function(e, n, f, i, $r, c, data) {
-                                        return /\b<%= escapeRegExp(tag.name) %>\b/.test(e);
-                                    },
-                                    <% }); %>
-                                },
-                                <% if (locals.course_instance) { %>
-                                5: {
-                                    <% all_assessments.forEach(function(assessment) { %>
-                                    "<%= escapeRegExp(assessment.label) %>": function(e, n, f, i, $r, c, data) {
-                                        return /\b<%= escapeRegExp(assessment.label) %>\b/.test(e);
-                                    },
-                                    <% }); %>
-                                },
-                                <% } %>
-                            },
-                        },
-                    });
-                });
-            </script>
+          </div>
+        </div>
+
+        <table id="questionsTable"
+               data-data="<%= JSON.stringify(questions) %>"
+               data-classes="table table-sm table-hover table-bordered"
+               data-thead-classes="thead-light"
+               data-filter-control="true"
+               data-show-filter-control-switch="true"
+               data-show-search-clear-button="true"
+               data-show-columns="true"
+               data-show-columns-toggle-all="true"
+               data-show-button-text="true"
+               data-pagination="true"
+               data-pagination-v-align="top"
+               data-page-list="[50,100,200,500,unlimited]"
+               data-page-size="50"
+               data-smart-display="false"
+               data-show-extended-pagination="true"
+               data-toolbar=".fixed-table-pagination"
+               data-sticky-header="true">
+          <thead>
+            <tr>
+              <th data-field="qid"
```

Comment:
Any chance of a tree view? 😄 (waiting for a future PR is completely reasonable)

Reply:
I remembered this a few seconds after submitting my comment, I'm working on it right now. Will let you know if it looks good.

Reply:
❤️

Reply:
treeview seems quite unstable from my limited tests. I'd prefer to move it to a separate PR where this can be revised more extensively.

Reply:
Very reasonable.